src/coreclr/vm/arm64/thunktemplates.S

```diff
@@ -120,9 +120,12 @@ LEAF_END_MARKED CallCountingStubCodeTemplate, _TEXT
     .irp STUB_PAGE_SIZE, 16384, 32768, 65536
 
     LEAF_ENTRY StubPrecodeCode\STUB_PAGE_SIZE
+        dmb ishld
```

Comment:
Do we need a counter-part barrier on the writer sides (e.g. in DynamicHelperFixup)?

Reply:
Yes. I had written... or at least thought about writing it. I'll have that put together soon.